nvi

```diff
@@ -100,24 +101,56 @@ function install_node() {
   mkdir --parents -- "$BIN_DIR"
 
   # Download and extract the tar if it doesn't exist on disk.
-  if [[ ! -d "$INSTALL_DIR/node-v$NODE_VERSION-linux-x64" ]]; then
+  if [[ ! -d "$INSTALL_DIR/$PACKAGE" ]]; then
     # Fail if version doesn't exist.
-    NODE_DIST_LOOKUP=$(wget -qO - https://nodejs.org/dist/index.json | grep --null-data --only-matching --perl-regex --text "\"version\"\s*:\s*\"v$NODE_VERSION\"")
-    [[ -z "$NODE_DIST_LOOKUP" ]] && >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." && exit 1
+    local NODE_DIST_LOOKUP=$(wget \
+      --quiet \
+      -O - \
+      https://nodejs.org/dist/index.json |
+    grep \
+      --null-data \
+      --only-matching \
+      --perl-regex \
+      --text \
+      "\"version\"\s*:\s*\"v$NODE_VERSION\"")
+    [[ -z "$NODE_DIST_LOOKUP" ]] &&
+      >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." &&
+      exit 1
     # Download and extract it if it does.
-    NODE_TAR="https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz"
-    if ! wget --output-document="$DOWNLOAD_DIR/$NODE_VERSION.tar.gz" --quiet "$NODE_TAR"; then
-      >&2 echo "Error: Couldn't download $NODE_TAR to ${PWD}/$NODE_VERSION.tar.gz, but the version is valid. Possibly a network or permission issue. Aborting." && exit 1
+    local ARCHIVE="$PACKAGE.tar.gz"
+    local DOWNLOAD_DEST="$DOWNLOAD_DIR/$ARCHIVE"
+    local NODE_TAR="https://nodejs.org/dist/v$NODE_VERSION/$ARCHIVE"
+    if ! wget \
+      --output-document="$DOWNLOAD_DEST" \
+      --quiet \
+      "$NODE_TAR"
+    then
+      >&2 echo "Error: Couldn't download $NODE_TAR to $DOWNLOAD_DEST, but the version is valid. Possibly a network or permission issue. Aborting." &&
+        exit 1
     fi
-    tar --extract --file "$DOWNLOAD_DIR/$NODE_VERSION.tar.gz" --gzip --directory "$INSTALL_DIR"
-    rm -- "$DOWNLOAD_DIR/$NODE_VERSION.tar.gz"
+    tar \
+      --extract \
+      --file \
+      "$DOWNLOAD_DEST" \
+      --gzip \
+      --directory \
+      "$INSTALL_DIR"
+    rm -- "$DOWNLOAD_DEST"
     # If the download dir was created by us and is now empty, remove it.
-    [[ ! -z $FRESH_DOWNLOAD_DIR ]] && [[ -z "$(ls -A "$DOWNLOAD_DIR")" ]] && rm -d -- "$DOWNLOAD_DIR"
+    [[ ! -z $FRESH_DOWNLOAD_DIR ]] &&
+      [[ -z "$(ls -A "$DOWNLOAD_DIR")" ]] &&
+      rm -d -- "$DOWNLOAD_DIR"
   fi
 
   # Install executables in the bin dir.
-  cp -- "$INSTALL_DIR/node-v$NODE_VERSION-linux-x64/bin/node" "$BIN_DIR/node"
-  ln --force --symbolic -- "$(readlink --canonicalize "$INSTALL_DIR/node-v$NODE_VERSION-linux-x64/bin/npm")" "$BIN_DIR/npm"
+  cp -- "$INSTALL_DIR/$PACKAGE/bin/node" "$BIN_DIR/node"
+  ln \
+    --force \
+    --symbolic \
+    -- \
+    "$(readlink --canonicalize \
+    "$INSTALL_DIR/$PACKAGE/bin/npm")" \
+    "$BIN_DIR/npm"
```

Comment:
Because `npm` is just a symlink when you download it. From my local system:

```
emil@botnet:~$ ./nvi -d node_down -e node_exec -i node_inst -n 7.5.0
emil@botnet:~$ cd node_inst/node-v7.5.0-linux-x64/bin
emil@botnet:~/node_inst/node-v7.5.0-linux-x64/bin$ ls -l
total 33720
-rwxrwxr-x 1 emil emil 34513043 Jan 31 22:12 node
lrwxrwxrwx 1 emil emil       38 Jan 31 22:12 npm -> ../lib/node_modules/npm/bin/npm-cli.js
```

I'm essentially trying to supply it in the format you'd normally find it in.